Libplanet.Action.Tests/Mocks/MockWorldState.cs

```diff
@@ -0,0 +1,42 @@
+using System.Collections.Immutable;
+using Libplanet.Action.State;
+using Libplanet.Crypto;
+using Libplanet.Store;
+using Libplanet.Store.Trie;
+
+namespace Libplanet.Action.Tests.Mocks
+{
+    public class MockWorldState : IWorldState
+    {
+        private readonly IImmutableDictionary<Address, IAccount> _accounts;
+
+        public MockWorldState()
+            : this(ImmutableDictionary<Address, IAccount>.Empty)
+        {
+        }
+
+        public MockWorldState(IImmutableDictionary<Address, IAccount> accounts)
+            : this(new TrieStateStore(new MemoryKeyValueStore()).GetStateRoot(null), accounts)
+        {
+        }
+
+        public MockWorldState(ITrie trie, IImmutableDictionary<Address, IAccount> accounts)
+        {
+            Trie = trie;
+            Legacy = true;
```

Comment:
Same with `World` in regards to its API. Seems like this is fixed to `true`. This should be able to emulate "legacy" and "modern". 😶

Reply:
Hmm, Do we really need `MockWorldState` class? May be we can use `WorldBaseState` on tests too for now.

Reply:
Maybe, its purpose is to easily mock an `IWorldState`. Mock accounts were introduced to bypass rather complicated logic implemented internally for `Account`s. I guess `World` is simple enough at this stage that we can simply use the actual implementation instead of a mock. 🙄

Reply:
Actually, I thought replacing `MockAccountState` too. Recent implementation of `IAccountState` holds `Trie`, but for Mock, it's a bit hard to make use of it. (To use it, this will be copy of `AcccountBaseState` I think)

Reply:
I agree, not enough thought has been put into `MockAccountState` and it isn't being used as widely as I originally wanted it to be. 😥

However, there is no easy way to mock a base state without actually manipulating `ITrie`, which then requires knowing how data is internally stored, which in turn requires to know how `KeyConverters` is implemented.

For practical application, when testing an `IAction.Execute()` for a client, there is no easy way to create a custom base state (mainly due to `Account` logic being overly complicated).